### PR TITLE
fixes mouse tooltip item value color not matching ground item value color

### LINKED ISSUE #11277: Mouse tooltip item value color does not match ground item value color
**Describe the bug**
When hovering over an item on the ground, the color of the mouse tooltip name of the item does not match the color of the text of the ground item. This only seems to happen on untradable items from my small amount of testing with feathers and cannonballs, though it is a small sample size.

**Screenshots**
NQcjIRbL5J
CyuvFuUbOP
My ground item settings for reference:
RuneLite_Zp40O7J4IQ

**Environment:**
 - OS: Windows 10
 - RuneLite version: 1.6.10.1
 - Launcher version: 1.6.0

### PR DESCRIPTION
**Pull request by rnoennig**
Fixes #11277

### CODE REVIEW

**Comment by raiyni** [comment]
You need to fix your imports according to our style guide. https://github.com/runelite/runelite/wiki/Code-Conventions

**Comment by raiyni** [comment]
My previous comment still applies.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 1]
```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
-			final int itemPrice = itemManager.getItemPrice(realItemId);
-			final int price = itemPrice <= 0 ? itemComposition.getPrice() : itemPrice;
-			final int haPrice = Math.round(itemComposition.getPrice() * Constants.HIGH_ALCHEMY_MULTIPLIER) * quantity;
-			final int gePrice = quantity * price;
-			final Color hidden = getHidden(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice, itemComposition.isTradeable());
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int itemPrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final int gePrice = itemPrice <= 0 ? haPrice : itemPrice;
+			final Color hidden = getHidden(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice, groundItem.isTradeable());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:505]: I don't understand why you have removed quantity from the color calculation here? Also, If we want to make this the same as the overlay color, can't we just copy the logic from the overlay? Which is:
```
			final Color highlighted = plugin.getHighlighted(new NamedQuantity(item), item.getGePrice(), item.getHaPrice());
			final Color hidden = plugin.getHidden(new NamedQuantity(item), item.getGePrice(), item.getHaPrice(), item.isTradeable());
```

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 2]
```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
-			final int realItemId = itemComposition.getNote() != -1 ? itemComposition.getLinkedNoteId() : itemComposition.getId();
-			final int itemPrice = itemManager.getItemPrice(realItemId);
-			final int price = itemPrice <= 0 ? itemComposition.getPrice() : itemPrice;
-			final int haPrice = Math.round(itemComposition.getPrice() * Constants.HIGH_ALCHEMY_MULTIPLIER) * quantity;
-			final int gePrice = quantity * price;
-			final Color hidden = getHidden(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice, itemComposition.isTradeable());
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int itemPrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final int gePrice = itemPrice <= 0 ? haPrice : itemPrice;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:504]: Why is this here? The overlay doesn't have that:

```java
		for (GroundItem item : groundItemList)
		{
			final LocalPoint groundPoint = LocalPoint.fromWorld(client, item.getLocation());

			if (groundPoint == null || localLocation.distanceTo(groundPoint) > MAX_DISTANCE
				|| (onlyShowLoot && !item.isMine()))
			{
				continue;
			}

			final Color highlighted = plugin.getHighlighted(new NamedQuantity(item), item.getGePrice(), item.getHaPrice());
```

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 3]
```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
+			final Color hidden = getHidden(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice, groundItem.isTradeable());
+			final Color highlighted = getHighlighted(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice);
 			final Color color = getItemColor(highlighted, hidden);
-			final boolean canBeRecolored = highlighted != null || (hidden != null && config.recolorMenuHiddenItems());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:520]: Why are you deleting this feature? It looks unrelated.

Comment (rnoennig) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:520]: Thanks for looking into this. This condition prevents that the computed color is actually applied if the value doesn't match any configured value and the default color should be applied.
Can you clarify which feature is actually removed by this edit? Thanks.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java (hunk 2 above)
Comment (rnoennig) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:504]: I assume you mean the ternary operator that uses the HA price when GE is 0? If not please clarify what you mean by "this".
I tried to preserve the existing logic as best as I could. It's possible that this can be removed, I'll test it and report back.

Please compare with the version from branch master:
https://github.com/runelite/runelite/blob/1c9c71ef7c93ccf4c479b7137635e5e266d07788/runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java#L514

Comment (rnoennig) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java]: This price defaulting is indeed not necessary. I removed it.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java (hunk 3 above)
Comment (rnoennig) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:520]: I didn't notice the feature "recolorMenuHiddenItems". I brought it back and edited this pull request.

**Comment by Nightfirecat** [comment]
It would help a lot for reviewing to squash this into two commits, one which fixes each of the issues you described in the issue.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java (hunk 3 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:520]: Why are you still removing this highlighted check? This will make `recolorMenuHiddenItems` apply to highlighted items, and also break items which are highlighted and hidden (which should show it as highlighted, as the overlay does)

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 4]
```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
-			final int gePrice = quantity * price;
-			final Color hidden = getHidden(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice, itemComposition.isTradeable());
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int gePrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final Color hidden = getHidden(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice, groundItem.isTradeable());
+			final Color highlighted = getHighlighted(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice);
 			final Color color = getItemColor(highlighted, hidden);
-			final boolean canBeRecolored = highlighted != null || (hidden != null && config.recolorMenuHiddenItems());
+			final boolean shouldNotRecolorHiddenMenuItem = hidden != null && !config.recolorMenuHiddenItems();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:507]: In general avoid double-negatives as it is hard to follow for me (eg `!shouldNot...)`

**Comment by Adam-** [comment]
Unrelated but it may be worth also writing a test for some of this highlight/hide/coloring behavior since it is not so straight forward

**Comment by rnoennig** [comment]
> Unrelated but it may be worth also writing a test for some of this highlight/hide/coloring behavior since it is not so straight forward

Good idea, I'll write a test for the features you mentioned and what was described in the original bug report.

**Comment by rnoennig** [comment]
> It would help a lot for reviewing to squash this into two commits, one which fixes each of the issues you described in the issue.

@Nightfirecat can I still do this within this PR or do I need to create a new one?

**Comment by dekvall** [comment]
You should do it within this PR.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java [hunk 5]
```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
...
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int gePrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final Color hidden = getHidden(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice, groundItem.isTradeable());
+			final Color highlighted = getHighlighted(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice);
 			final Color color = getItemColor(highlighted, hidden);
 			final boolean canBeRecolored = highlighted != null || (hidden != null && config.recolorMenuHiddenItems());
 
-			if (color != null && canBeRecolored && !color.equals(config.defaultColor()))
+			if (color != null && canBeRecolored)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java:509]: Why is this being removed? It would cause the plugin to apply the default overlay color to the menus which it currently doesn't do. It seems unrelated to this bug at least. I think the reason this exists is because the overlay needs a color but the menus already have their own default color which is white.

File: runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+	@Bind
+	private Notifier notifier;
+
+	@Mock
+	@Bind
+	private ScheduledExecutorService executor;
+
+	@Mock
+	@Bind
+	private RuneLiteConfig runeLiteConfig;
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java:113]: This isn't injected into the plugin, so it probably doesn't need to be mock/bound?

File: runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+		when(tileItem.getId()).thenReturn(itemId);
+		when(tile.getItemLayer()).thenReturn(itemLayer);
+		when(itemManager.getItemComposition(itemId)).thenReturn(itemComposition);
+		when(client.getLocalPlayer()).thenReturn(player);
+		when(player.getWorldLocation()).thenReturn(worldPoint);
+		when(tile.getWorldLocation()).thenReturn(worldPoint);
+		when(itemComposition.getNote()).thenReturn(-1);
+		when(client.getMenuEntries()).thenReturn(new MenuEntry[]{menuEntry});
+		when(client.getScene()).thenReturn(scene);
+		when(scene.getTiles()).thenReturn(new Tile[][][]{new Tile[][]{new Tile[]{tile}}});
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java:219]: Instead of mocking all of this code, you can just remove it from the plugin, since it is unused

**Comment by Adam-** [comment]
I ended up just fixing the last part myself. The test wasn't really what I had envisioned so I removed it.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java (hunk 5 above)
Comment (rnoennig) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java]: If I understood your comment correctly, the default color should only be used for the overlay and not for the menu entry? I recommend changing it so both match the configured colors(highlight, default, hidden). Otherwise I'd recommend changing the description of the color so that's it's clear that the default color doesn't change the default color of menu entries.

File: runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java (hunk 7 above)
Comment (rnoennig) [reply runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java]: I'm not sure what you mean. The 4 lines your highlighted are non-lenient stubs, if they aren't called during execution an error would be thrown. I assure you those lines are needed.

**Comment by rnoennig** [comment]
> I ended up just fixing the last part myself. The test wasn't really what I had envisioned so I removed it.

@Adam- What had you envisioned?

File: runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java (hunk 7 above)
Comment (Adam-) [reply runelite-client/src/test/java/net/runelite/client/plugins/grounditems/GroundItemsPluginTest.java]: If you look at the final commit I pushed, it removed the code which used this from the plugin, which makes it unnecessary.

File: runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java (hunk 5 above)
Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java]: This would cause a rather user visible behavior change, which generally if we do those (especially without a config option toggle) it upsets people. At the very least it is not related to the two bugs this is trying to fix though.
